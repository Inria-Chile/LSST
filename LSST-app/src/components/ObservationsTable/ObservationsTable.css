.ReactTable .rt-noData {
    padding: 0px 5px;
}

.ReactTable {
    /* border: 1px solid #2d4053; */
    border: 1px solid rgba(0,0,0,0.0);
    color: #d4d7da;
}

.rt-thead {
    background: #5b6b7b;
}

.rt-th {
    text-align: left;
    margin-left: 10px;
    font-weight: bold;
    min-height: 35px;
    vertical-align: center;
}

.rt-td {
    margin-left: 10px;
}

.rt-tr-group {
    margin-top: 4px;
    background: #435568;
}

.-no-header > .rt-table > .rt-thead{
    display: none;
}

.rt-resizable-header-content {
    position: relative;
    top: 50%;
    transform: translateY(-50%);
}

.scrollbar > div:nth-child(3){
    background: rgba(0,0,0,0.0);
    margin: 10px 3px;
    width: 10px !important;
}

.scrollbar > div:nth-child(3) > div{
    background: #5b6b7b;
    background-color: #5b6b7b !important;
    border-radius: 10px !important;
}