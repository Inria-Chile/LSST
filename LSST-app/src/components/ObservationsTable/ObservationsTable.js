import React, { PureComponent } from 'react';

// Import React Table
import ReactTable from "react-table";
import "react-table/react-table.css";
import './ObservationsTable.css';
import { filterColors, lstToscienceProposal, decreaseBrightness } from "../Utils/Utils";
import { Scrollbars } from "react-custom-scrollbars";

class ObservationsTable extends PureComponent {

    constructor() {
        super();
        this.state = {
            clickedData: [{}]
        };
    }

    render() {
        let timestampColWidth = 150;
        let filterColWidth = 120;
        return (
            <div className="observations-table-wrapper">
                <div style={{height: '35px'}}>
                <ReactTable
                        sortable={false}
                        data={[]}
                        columns={[
                            {
                                Header: "Timestamp",
                                id: "expDate",
                                maxWidth: timestampColWidth,
                                headerStyle: {textAlign: 'left'}
                            },
                            {
                                Header: "Filter",
                                maxWidth: filterColWidth,
                                headerStyle: {textAlign: 'left'}
                            },
                            {
                                Header: "Science proposals",
                            }
                        ]}
                        defaultPageSize={0}
                        pageSize={0}
                        className="-striped -highlight"
                        showPaginationBottom={false}
                        noDataText=""
                        
                    />
                </div>
                <div style={{height: 'calc(100% - 35px)'}}>
                <Scrollbars className='scrollbar'
                    style={{ height: '100%' }}>
                    <ReactTable
                        sortable={false}
                        data={this.props.clickedField === null ? [] : [...this.props.clickedField]}
                        columns={[
                            {
                                Header: "Timestamp",
                                id: "expDate",
                                accessor: d => d.expDate,
                                maxWidth: timestampColWidth
                            },
                            {
                                Header: "Filter",
                                accessor: "filterName",
                                Cell: row => (
                                        <div style={{
                                            width: '80%',
                                            backgroundColor: filterColors[row.value ] ? filterColors[row.value ]
                                                : "#000000",
                                            textAlign: "center",
                                            borderRadius: '2px',
                                            border: decreaseBrightness(filterColors[row.value], 1.3) + ' solid 2px'
                                        }}>
                                        {
                                            row.value 
                                        }
                                        </div> 
                                ),
                                maxWidth: filterColWidth
                            },
                            {
                                Header: "Science proposals",
                                id: "lst",
                                accessor: d => d.lst ? lstToscienceProposal(d.lst) : ''
                            }
                        ]}
                        defaultPageSize={9}
                        pageSize={this.props.clickedField === null || this.props.clickedField.length < 9 ? 9 : this.props.clickedField.length}
                        className="-highlight -no-header"
                        showPaginationBottom={false}
                        noDataText=""
                        
                    />
                </Scrollbars>
                </div>
            </div>
        );
    }
}

export default ObservationsTable;