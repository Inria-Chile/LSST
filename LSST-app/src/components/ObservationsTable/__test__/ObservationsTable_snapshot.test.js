import React from 'react';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';
import ObservationsTable from '../ObservationsTable';



/*---- Snapshot test ----*/
describe('ObservationsTable_Test',function(){
    it('renders correctly with shallowRender',()=> {
        const clickedField = ['x','y','z'];
        const renderer = new shallowRenderer();
        const tree = renderer.render(
          <ObservationsTable clickedField = {clickedField} />);
          expect(tree).toMatchSnapshot();
      });
    
      it('renders correctly without clickedField setted',()=> {
        const clickedField = ['x','y','z'];
        const renderer = new shallowRenderer();
        const tree = renderer.render(
          <ObservationsTable clickedField = {null} />);
          expect(tree).toMatchSnapshot();
      });
});