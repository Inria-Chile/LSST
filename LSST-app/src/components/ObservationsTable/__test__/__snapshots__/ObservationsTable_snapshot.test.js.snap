// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationsTable_Test renders correctly with shallowRender 1`] = `
<div
  className="observations-table-wrapper"
>
  <div
    style={
      Object {
        "height": "35px",
      }
    }
  >
    <ReactTable
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NextComponent={undefined}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotComponent={undefined}
      PivotValueComponent={[Function]}
      PreviousComponent={undefined}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="-striped -highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Header": "Timestamp",
            "headerStyle": Object {
              "textAlign": "left",
            },
            "id": "expDate",
            "maxWidth": 250,
          },
          Object {
            "Header": "Filter",
            "headerStyle": Object {
              "textAlign": "left",
            },
            "maxWidth": 120,
          },
          Object {
            "Header": "Science proposals",
          },
        ]
      }
      data={Array []}
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPageSize={0}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={Array []}
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Loading..."
      multiSort={true}
      nestingLevelKey="_nestingLevel"
      nextText="Next"
      noDataText=""
      ofText="of"
      onExpandedChange={undefined}
      onFetchData={[Function]}
      onFilteredChange={undefined}
      onPageChange={undefined}
      onPageSizeChange={undefined}
      onResizedChange={undefined}
      onSortedChange={undefined}
      originalKey="_original"
      pageSize={0}
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Page"
      pivotBy={undefined}
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Previous"
      resizable={true}
      rowsText="rows"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={true}
      showPaginationBottom={false}
      showPaginationTop={false}
      sortable={false}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
  <div
    style={
      Object {
        "height": "calc(100% - 35px)",
      }
    }
  >
    <Scrollbars
      autoHeight={false}
      autoHeightMax={200}
      autoHeightMin={0}
      autoHide={false}
      autoHideDuration={200}
      autoHideTimeout={1000}
      className="scrollbar"
      hideTracksWhenNotNeeded={false}
      renderThumbHorizontal={[Function]}
      renderThumbVertical={[Function]}
      renderTrackHorizontal={[Function]}
      renderTrackVertical={[Function]}
      renderView={[Function]}
      style={
        Object {
          "height": "100%",
        }
      }
      tagName="div"
      thumbMinSize={30}
      universal={false}
    >
      <ReactTable
        AggregatedComponent={[Function]}
        ExpanderComponent={[Function]}
        FilterComponent={[Function]}
        LoadingComponent={[Function]}
        NextComponent={undefined}
        NoDataComponent={[Function]}
        PadRowComponent={[Function]}
        PaginationComponent={[Function]}
        PivotComponent={undefined}
        PivotValueComponent={[Function]}
        PreviousComponent={undefined}
        ResizerComponent={[Function]}
        TableComponent={[Function]}
        TbodyComponent={[Function]}
        TdComponent={[Function]}
        TfootComponent={[Function]}
        ThComponent={[Function]}
        TheadComponent={[Function]}
        TrComponent={[Function]}
        TrGroupComponent={[Function]}
        aggregatedKey="_aggregated"
        className="-highlight -no-header"
        collapseOnDataChange={true}
        collapseOnPageChange={true}
        collapseOnSortingChange={true}
        column={
          Object {
            "Aggregated": undefined,
            "Cell": undefined,
            "Expander": undefined,
            "Filter": undefined,
            "Footer": undefined,
            "Header": undefined,
            "Pivot": undefined,
            "PivotValue": undefined,
            "aggregate": undefined,
            "className": "",
            "filterAll": false,
            "filterMethod": undefined,
            "filterable": undefined,
            "footerClassName": "",
            "footerStyle": Object {},
            "getFooterProps": [Function],
            "getHeaderProps": [Function],
            "getProps": [Function],
            "headerClassName": "",
            "headerStyle": Object {},
            "minWidth": 100,
            "resizable": undefined,
            "show": true,
            "sortMethod": undefined,
            "sortable": undefined,
            "style": Object {},
          }
        }
        columns={
          Array [
            Object {
              "Header": "Timestamp",
              "accessor": [Function],
              "id": "expDate",
              "maxWidth": 250,
            },
            Object {
              "Cell": [Function],
              "Header": "Filter",
              "accessor": "filterName",
              "maxWidth": 120,
            },
            Object {
              "Header": "Science proposals",
              "accessor": [Function],
              "id": "lst",
            },
          ]
        }
        data={
          Array [
            "x",
            "y",
            "z",
          ]
        }
        defaultExpanded={Object {}}
        defaultFilterMethod={[Function]}
        defaultFiltered={Array []}
        defaultPageSize={11}
        defaultResized={Array []}
        defaultSortDesc={false}
        defaultSortMethod={[Function]}
        defaultSorted={Array []}
        expanderDefaults={
          Object {
            "filterable": false,
            "resizable": false,
            "sortable": false,
            "width": 35,
          }
        }
        filterable={false}
        freezeWhenExpanded={false}
        getLoadingProps={[Function]}
        getNoDataProps={[Function]}
        getPaginationProps={[Function]}
        getProps={[Function]}
        getResizerProps={[Function]}
        getTableProps={[Function]}
        getTbodyProps={[Function]}
        getTdProps={[Function]}
        getTfootProps={[Function]}
        getTfootTdProps={[Function]}
        getTfootTrProps={[Function]}
        getTheadFilterProps={[Function]}
        getTheadFilterThProps={[Function]}
        getTheadFilterTrProps={[Function]}
        getTheadGroupProps={[Function]}
        getTheadGroupThProps={[Function]}
        getTheadGroupTrProps={[Function]}
        getTheadProps={[Function]}
        getTheadThProps={[Function]}
        getTheadTrProps={[Function]}
        getTrGroupProps={[Function]}
        getTrProps={[Function]}
        groupedByPivotKey="_groupedByPivot"
        indexKey="_index"
        loading={false}
        loadingText="Loading..."
        multiSort={true}
        nestingLevelKey="_nestingLevel"
        nextText="Next"
        noDataText=""
        ofText="of"
        onExpandedChange={undefined}
        onFetchData={[Function]}
        onFilteredChange={undefined}
        onPageChange={undefined}
        onPageSizeChange={undefined}
        onResizedChange={undefined}
        onSortedChange={undefined}
        originalKey="_original"
        pageSize={10}
        pageSizeOptions={
          Array [
            5,
            10,
            20,
            25,
            50,
            100,
          ]
        }
        pageText="Page"
        pivotBy={undefined}
        pivotDefaults={Object {}}
        pivotIDKey="_pivotID"
        pivotValKey="_pivotVal"
        previousText="Previous"
        resizable={true}
        rowsText="rows"
        showPageJump={true}
        showPageSizeOptions={true}
        showPagination={true}
        showPaginationBottom={false}
        showPaginationTop={false}
        sortable={false}
        style={Object {}}
        subRowsKey="_subRows"
      />
    </Scrollbars>
  </div>
</div>
`;

exports[`ObservationsTable_Test renders correctly without clickedField setted 1`] = `
<div
  className="observations-table-wrapper"
>
  <div
    style={
      Object {
        "height": "35px",
      }
    }
  >
    <ReactTable
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NextComponent={undefined}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotComponent={undefined}
      PivotValueComponent={[Function]}
      PreviousComponent={undefined}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="-striped -highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Header": "Timestamp",
            "headerStyle": Object {
              "textAlign": "left",
            },
            "id": "expDate",
            "maxWidth": 250,
          },
          Object {
            "Header": "Filter",
            "headerStyle": Object {
              "textAlign": "left",
            },
            "maxWidth": 120,
          },
          Object {
            "Header": "Science proposals",
          },
        ]
      }
      data={Array []}
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPageSize={0}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={Array []}
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Loading..."
      multiSort={true}
      nestingLevelKey="_nestingLevel"
      nextText="Next"
      noDataText=""
      ofText="of"
      onExpandedChange={undefined}
      onFetchData={[Function]}
      onFilteredChange={undefined}
      onPageChange={undefined}
      onPageSizeChange={undefined}
      onResizedChange={undefined}
      onSortedChange={undefined}
      originalKey="_original"
      pageSize={0}
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Page"
      pivotBy={undefined}
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Previous"
      resizable={true}
      rowsText="rows"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={true}
      showPaginationBottom={false}
      showPaginationTop={false}
      sortable={false}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
  <div
    style={
      Object {
        "height": "calc(100% - 35px)",
      }
    }
  >
    <Scrollbars
      autoHeight={false}
      autoHeightMax={200}
      autoHeightMin={0}
      autoHide={false}
      autoHideDuration={200}
      autoHideTimeout={1000}
      className="scrollbar"
      hideTracksWhenNotNeeded={false}
      renderThumbHorizontal={[Function]}
      renderThumbVertical={[Function]}
      renderTrackHorizontal={[Function]}
      renderTrackVertical={[Function]}
      renderView={[Function]}
      style={
        Object {
          "height": "100%",
        }
      }
      tagName="div"
      thumbMinSize={30}
      universal={false}
    >
      <ReactTable
        AggregatedComponent={[Function]}
        ExpanderComponent={[Function]}
        FilterComponent={[Function]}
        LoadingComponent={[Function]}
        NextComponent={undefined}
        NoDataComponent={[Function]}
        PadRowComponent={[Function]}
        PaginationComponent={[Function]}
        PivotComponent={undefined}
        PivotValueComponent={[Function]}
        PreviousComponent={undefined}
        ResizerComponent={[Function]}
        TableComponent={[Function]}
        TbodyComponent={[Function]}
        TdComponent={[Function]}
        TfootComponent={[Function]}
        ThComponent={[Function]}
        TheadComponent={[Function]}
        TrComponent={[Function]}
        TrGroupComponent={[Function]}
        aggregatedKey="_aggregated"
        className="-highlight -no-header"
        collapseOnDataChange={true}
        collapseOnPageChange={true}
        collapseOnSortingChange={true}
        column={
          Object {
            "Aggregated": undefined,
            "Cell": undefined,
            "Expander": undefined,
            "Filter": undefined,
            "Footer": undefined,
            "Header": undefined,
            "Pivot": undefined,
            "PivotValue": undefined,
            "aggregate": undefined,
            "className": "",
            "filterAll": false,
            "filterMethod": undefined,
            "filterable": undefined,
            "footerClassName": "",
            "footerStyle": Object {},
            "getFooterProps": [Function],
            "getHeaderProps": [Function],
            "getProps": [Function],
            "headerClassName": "",
            "headerStyle": Object {},
            "minWidth": 100,
            "resizable": undefined,
            "show": true,
            "sortMethod": undefined,
            "sortable": undefined,
            "style": Object {},
          }
        }
        columns={
          Array [
            Object {
              "Header": "Timestamp",
              "accessor": [Function],
              "id": "expDate",
              "maxWidth": 250,
            },
            Object {
              "Cell": [Function],
              "Header": "Filter",
              "accessor": "filterName",
              "maxWidth": 120,
            },
            Object {
              "Header": "Science proposals",
              "accessor": [Function],
              "id": "lst",
            },
          ]
        }
        data={Array []}
        defaultExpanded={Object {}}
        defaultFilterMethod={[Function]}
        defaultFiltered={Array []}
        defaultPageSize={11}
        defaultResized={Array []}
        defaultSortDesc={false}
        defaultSortMethod={[Function]}
        defaultSorted={Array []}
        expanderDefaults={
          Object {
            "filterable": false,
            "resizable": false,
            "sortable": false,
            "width": 35,
          }
        }
        filterable={false}
        freezeWhenExpanded={false}
        getLoadingProps={[Function]}
        getNoDataProps={[Function]}
        getPaginationProps={[Function]}
        getProps={[Function]}
        getResizerProps={[Function]}
        getTableProps={[Function]}
        getTbodyProps={[Function]}
        getTdProps={[Function]}
        getTfootProps={[Function]}
        getTfootTdProps={[Function]}
        getTfootTrProps={[Function]}
        getTheadFilterProps={[Function]}
        getTheadFilterThProps={[Function]}
        getTheadFilterTrProps={[Function]}
        getTheadGroupProps={[Function]}
        getTheadGroupThProps={[Function]}
        getTheadGroupTrProps={[Function]}
        getTheadProps={[Function]}
        getTheadThProps={[Function]}
        getTheadTrProps={[Function]}
        getTrGroupProps={[Function]}
        getTrProps={[Function]}
        groupedByPivotKey="_groupedByPivot"
        indexKey="_index"
        loading={false}
        loadingText="Loading..."
        multiSort={true}
        nestingLevelKey="_nestingLevel"
        nextText="Next"
        noDataText=""
        ofText="of"
        onExpandedChange={undefined}
        onFetchData={[Function]}
        onFilteredChange={undefined}
        onPageChange={undefined}
        onPageSizeChange={undefined}
        onResizedChange={undefined}
        onSortedChange={undefined}
        originalKey="_original"
        pageSize={10}
        pageSizeOptions={
          Array [
            5,
            10,
            20,
            25,
            50,
            100,
          ]
        }
        pageText="Page"
        pivotBy={undefined}
        pivotDefaults={Object {}}
        pivotIDKey="_pivotID"
        pivotValKey="_pivotVal"
        previousText="Previous"
        resizable={true}
        rowsText="rows"
        showPageJump={true}
        showPageSizeOptions={true}
        showPagination={true}
        showPaginationBottom={false}
        showPaginationTop={false}
        sortable={false}
        style={Object {}}
        subRowsKey="_subRows"
      />
    </Scrollbars>
  </div>
</div>
`;
